react_ujs/src/renderComponent/withReactDOM.js

```diff
@@ -0,0 +1,10 @@
+var ReactDOM = require("react-dom")
+
+// Render React component via ReactDOM, for example:
+//
+// - `renderComponent("hydrate", component, node, props)` -> `ReactDOM.hydrate(component, node);`
+// - `renderComponent("render", component, node, props)` -> `ReactDOM.render(component, node);`
+//
+module.exports = function(renderFunctionName, component, node) {
+  ReactDOM[renderFunctionName](component, node);
+};
```

Comment:
The last argument `props` here is unused in this default render function. However, it is provided to support extensibility for custom implementations. The hot reload variation, for uses `props` to re-render with a freshly-reloaded component constructor.

Let me know if
1. function should declare the `props` argument even if it's unused (for self-documentation/ease of copy-and-pasting)
1. function signature should change to pass `constructor` instead of/in addition to `component`. The existing implementation of cached turbolinks components would need extra considerations if `component` is not passed to this function

Previous proposal referenced: https://github.com/reactjs/react-rails/pull/595#issuecomment-243638753

> - from
> ```js
> ReactDOM.render(React.createElement(constructor, props), node);
> ```
> - to
> ```js
> ReactRailsUJS.renderComponent(node, constructor, props)
> 
> // .... then ... 
> renderComponent: function(node, constructor, props) { 
>   ReactDOM.render(React.createElement(constructor, props), node);
> },
> ```